packages/api/src/graphql/comment.ts

```diff
@@ -0,0 +1,153 @@
+import {Context} from '../context'
+import {
+  CommentAuthorType,
+  CommentItemType,
+  CommentRejectionReason,
+  CommentRevision,
+  CommentStatus,
+  PublicComment
+} from '../db/comment'
+import {GraphQLPageInfo} from './common'
+import {GraphQLRichText} from './richText'
+
+export const GraphQLCommentStatus = new GraphQLEnumType({
+  name: 'CommentStatus',
+  values: {
+    [CommentStatus.Approved]: {value: CommentStatus.Approved},
+    [CommentStatus.PendingApproval]: {value: CommentStatus.PendingApproval},
+    [CommentStatus.PendingUserChanges]: {value: CommentStatus.PendingUserChanges},
+    [CommentStatus.Rejected]: {value: CommentStatus.Rejected}
+  }
+})
+
+export const GraphQLCommentRejectionReason = new GraphQLEnumType({
+  name: 'CommentRejectionReason',
+  values: {
+    [CommentRejectionReason.Misconduct]: {value: CommentRejectionReason.Misconduct},
+    [CommentRejectionReason.Spam]: {value: CommentRejectionReason.Spam}
+  }
+})
+
+export const GraphQLCommentAuthorType = new GraphQLEnumType({
+  name: 'CommentAuthorType',
+  values: {
+    [CommentAuthorType.Author]: {value: CommentAuthorType.Author},
+    [CommentAuthorType.Team]: {value: CommentAuthorType.Team},
+    [CommentAuthorType.VerifiedUser]: {value: CommentAuthorType.VerifiedUser}
+  }
+})
+
+export const GraphQLCommentItemType = new GraphQLEnumType({
+  name: 'CommentItemType',
+  values: {
+    [CommentItemType.Article]: {value: CommentItemType.Article},
+    [CommentItemType.Page]: {value: CommentItemType.Page}
+  }
+})
+
+export const GraphQLCommentFilter = new GraphQLInputObjectType({
+  name: 'CommentFilter',
+  fields: {
+    status: {type: GraphQLCommentStatus}
+  }
+})
+
+export const GraphQLCommentRevision = new GraphQLObjectType<CommentRevision, Context>({
+  name: 'CommentRevision',
+  fields: {
+    text: {type: GraphQLNonNull(GraphQLRichText)},
+    createdAt: {type: GraphQLNonNull(GraphQLDateTime)}
+  }
+})
+
+export const GraphQLCommentRevisionInput = new GraphQLInputObjectType({
+  name: 'CommentRevisionInput',
+  fields: {
+    text: {type: GraphQLNonNull(GraphQLRichText)}
+  }
+})
+
+export const GraphQLCommentInput = new GraphQLInputObjectType({
+  name: 'CommentInput',
+  fields: {
+    userID: {type: GraphQLNonNull(GraphQLID)},
+
+    itemID: {type: GraphQLNonNull(GraphQLID)},
+    itemType: {
+      type: GraphQLNonNull(GraphQLCommentItemType),
+      defaultValue: GraphQLCommentStatus.getValue(CommentItemType.Article)
+    },
+
+    revisions: {type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLCommentRevisionInput)))},
+    parentID: {type: GraphQLID},
+
+    status: {
+      type: GraphQLNonNull(GraphQLCommentStatus),
+      defaultValue: GraphQLCommentStatus.getValue(CommentStatus.PendingApproval)
+    },
+    rejectionReason: {type: GraphQLCommentRejectionReason},
+    authorType: {type: GraphQLNonNull(GraphQLCommentAuthorType)}
+  }
+})
+
+export const GraphQLComment = new GraphQLObjectType<Comment, Context>({
```

Comment:
I'm pretty sure you don't need that in the private API. There is no need for moderators to change comments through the editor.

Reply:
I am using it for private queries, what do you think?